CDCgov/prime-reportstream
# feat: new upload page [experience-6111]

Creates a page that can be used to upload csv or hl7 files into the reportStream pipeline.

The page is set behind a `user-upload` feature flag, and accessible to admins only.

This change includes a change to the navigation that will display a dropdown containing the new user upload page, and the existing validation page under a `Validate & Upload` nav link.

To support this, some things were added to file handler support components:
* more robust success and error messaging for upload vs validate
* support for timestamp and report id metadata display on success
* support for a warning banner on the upload page

Some idea of designs is here, though these will not match precisely https://www.figma.com/file/b2Zd1xzmp4cdwFO8O7edcI/CSV-Uploader-%26-Validation?node-id=2288%3A1127

## Test Steps:

1. start servers on this branch with `yarn run start:localdev` from `frontend-react` and `./gradlew run` from `prime-router`
2. visit http://localhost:3000/admin/settings and log in as necessary
4. activate the `user-upload` feature flag via the `admin -> feature flags` navigation
5. activate the `validation-service` feature flag via the `admin -> feature flags` navigation
6. refresh
7. _VERIFY_: you see the `Upload & Validate` dropdown with a `User Upload` and `Validate` links inside
3. set up your user up as a sender that is set up to send csv files, ex. `all-in-one-health-ca`
8. click `User Upload` in the nav bar under `Upload & Validate`
9. _VERIFY_: page looks like `nav` screenshot below. (Unfortunately Figma is not up to date with all of the content content here, but for the most part the design should match what the validator is doing.)
10.  upload a valid csv file
7. _VERIFY_: page looks like `upload success` screenshot below
8. upload an invalid csv file
11. _VERIFY_: page looks like `upload error` screenshot below
8. click `Validate` in the nav bar under `Upload & Validate`
10.  upload a valid csv file
7. _VERIFY_: page looks like `validation success` screenshot below
8. upload an invalid csv file
11. _VERIFY_: page looks like `validation error` screenshot below
12. refresh
13. upload a csv file
11. _VERIFY_: page looks like `server error` screenshot below

## Screenshots

Nav: 
<img width="1385" alt="Screen Shot 2022-07-19 at 2 52 43 PM" src="https://user-images.githubusercontent.com/92806979/179832243-ee979d0a-dcaf-4364-831b-f3578bc212ce.png">


Upload warning banner:
<img width="992" alt="Screen Shot 2022-07-19 at 11 10 51 AM" src="https://user-images.githubusercontent.com/92806979/179824255-0d75c6e4-7ebd-4411-a6e7-b93f05723df4.png">

Validator success:
<img width="1353" alt="Screen Shot 2022-07-19 at 10 20 57 AM" src="https://user-images.githubusercontent.com/92806979/179824260-d24699c0-47f0-412a-86d3-441a28fe72cb.png">

Upload success:
<img width="1318" alt="Screen Shot 2022-07-19 at 2 48 25 PM" src="https://user-images.githubusercontent.com/92806979/179826437-a373cf84-fb76-4c73-84c4-c2786d79e1a8.png">

Upload Error:
<img width="1362" alt="Screen Shot 2022-07-19 at 2 48 59 PM" src="https://user-images.githubusercontent.com/92806979/179826463-4ba457e3-5770-4139-b90d-f1cda9ceb8a8.png">

Server error:
<img width="1358" alt="Screen Shot 2022-07-19 at 10 18 13 AM" src="https://user-images.githubusercontent.com/92806979/179824268-4fe10428-d678-4590-95a7-920eb24f92f0.png">

Validation error:
<img width="1350" alt="Screen Shot 2022-07-19 at 10 17 22 AM" src="https://user-images.githubusercontent.com/92806979/179824271-e12f4148-2753-479f-8f22-9d37cc20d561.png">

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/cdcgov/prime-reportstream/6111
